src/lightning/data/streaming/cache.py

```diff
@@ -94,6 +94,10 @@ def __init__(
         self._is_done = False
         self._distributed_env = _DistributedEnv.detect()
 
+    @property
+    def rank(self) -> int:
+        return self._reader.rank
```

Comment:
Why would this make sense? The rank is something global, accessing it through a particular object doesn't make much sense. We should always use the distributed env for this. It's the source of truth.

Reply:
Done !